src/components/graph/ExerciseGraphNode.ts

```diff
@@ -0,0 +1,23 @@
+import { uuid } from './graph-types'
+
+export class ExerciseGraphNode {
+  public index: number
+  public slug: string
+  public uuid: uuid
+  public concepts: Array<string>
+  public prerequisites: Array<string>
+
+  constructor(
+    index: number,
+    slug: string,
+    uuid: uuid,
+    concepts: Array<string>,
+    prerequisites: Array<string>
+  ) {
+    this.index = index
+    this.slug = slug
+    this.uuid = uuid
+    this.concepts = [...concepts]
```

Comment:
```suggestion
    this.concepts = [...concepts]
```

Do you copy these just because? My guess is that there are side-effects somewhere else that should be resolved instead of copying here.